testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -3769,6 +3766,88 @@ protected void testColumnName(String columnName, boolean delimited)
         }
     }
 
+    @Test(dataProvider = "testColumnNameDataProvider")
+    public void testAddAndDropColumnName(String columnName)
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_ADD_COLUMN) && hasBehavior(SUPPORTS_DROP_COLUMN));
+
+        if (!requiresDelimiting(columnName)) {
+            testAddAndDropColumnName(columnName, false);
+        }
+        testAddAndDropColumnName(columnName, true);
+    }
+
+    protected void testAddAndDropColumnName(String columnName, boolean delimited)
+    {
+        String nameInSql = toColumnNameInSql(columnName, delimited);
+        String tableName = "tcn_" + nameInSql.toLowerCase(ENGLISH).replaceAll("[^a-z0-9]", "") + randomTableSuffix();
+
+        try {
+            assertUpdate("CREATE TABLE " + tableName + "(" + nameInSql + " varchar(50), value varchar(50))");
+        }
+        catch (RuntimeException e) {
+            if (isColumnNameRejected(e, columnName, delimited)) {
+                // It is OK if give column name is not allowed and is clearly rejected by the connector.
+                return;
+            }
+            throw e;
+        }
+        assertTableColumnNames(tableName, columnName.toLowerCase(ENGLISH), "value");
+
+        assertUpdate("ALTER TABLE " + tableName + " DROP COLUMN " + nameInSql);
+        assertTableColumnNames(tableName, "value");
+
+        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN " + nameInSql + " varchar(50)");
+        assertTableColumnNames(tableName, "value", columnName.toLowerCase(ENGLISH));
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test(dataProvider = "testColumnNameDataProvider")
+    public void testRenameColumnName(String columnName)
```

Comment:
Why do we need this new method `testRenameColumnName`? Why cannot we use/update old one `testRenameColumn`? What is the principal difference between them? How the name of these methods describe that difference?

Reply:
`testRenameColumn` should execute some operations to verify `RENAME COLUMN` works correctly (e.g. data didn't loss after renaming). `testRenameColumnName` should focus on test for column names.

> How the name of these methods describe that difference?

Please feel free to send PR if you feel the name isn't clear,